packages/admin-panel/src/VizBuilderApp/api/queries/useUser.js

```diff
@@ -0,0 +1,22 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+import { useQuery } from 'react-query';
+import { get } from '../api';
+
+export const useUser = options => {
+  const query = useQuery('user', () => get('user'), {
+    retry: 0,
+    // should be refetched in the background every hour
+    staleTime: 1000 * 60 * 60 * 1,
+    refetchOnWindowFocus: false,
+    ...options,
+  });
+
+  const user = query.data;
+  const isLoggedIn = user && Object.keys(user).length > 0;
+  const isBESAdmin = user && user.isBESAdmin;
+
+  return { ...query, isLoggedIn, isBESAdmin };
```

Comment:
One suggested refactor is to keep the `isBESAdmin` prop in the `user` object, since we generally pass both of them in the same components. That's not just for convenience but it probably makes sense conceptually

Reply:
I'm don't feel too strongly about it but it is a bit more convenient to have isBESAdmin outside the user object to check for access before showing user details. It's kind of user meta data so I think it's ok to be seperate.

Reply:
I see what you mean - `user` may be `undefined`... Fair enough 🙂